openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -1683,6 +1683,16 @@ def test_register_duplicate_username_and_email(self):
             }
         )
 
+    @override_settings(REGISTRATION_EXTRA_FIELDS={"honor_code": "hidden", "terms_of_service": "hidden"})
+    def test_register_hidden_honor_code_and_terms_of_service(self):
```

Comment:
If honor_code is set to `required` (the default), a combined honor code + terms of service checkbox is displayed. The view then [copies the `honor_code` parameter to the `terms_of_service` parameter](https://github.com/open-craft/edx-platform/blob/d114be732f69cd87289b68375207de4deeea1675/openedx/core/djangoapps/user_api/views.py#L322-L330) before passing the POST data to the form for validation.

This change fixes the case when *both* checkboxes are hidden, which does not currently work.